Compiling on nvidia tegra tk1 / jetson tk1 board

[ 66%] Building CXX object build_picongpu/build_mpiInfo/CMakeFiles/mpiInfo.dir/main.cpp.o
/home/ubuntu/code/picongpu/src/cuda_memtest/tests.cu(227): warning: shift count is too large

/home/ubuntu/code/picongpu/src/cuda_memtest/tests.cu(227): warning: shift count is too large

Linking CXX executable mpiInfo
[ 66%] Built target mpiInfo
/home/ubuntu/code/picongpu/src/cuda_memtest/cuda_memtest.cu: In function ‘void display_device_info(cudaDeviceProp*)’:
/home/ubuntu/code/picongpu/src/cuda_memtest/cuda_memtest.cu:113:212: warning: format ‘%llu’ expects argument of type ‘long long unsigned int’, but argument 7 has type ‘size_t {aka unsigned int}’ [-Wformat=]
     PRINTF("Device name=%s, global memory size=%llu\n", prop->name, prop->totalGlobalMem);
                                                                                                                                                                                                                    ^
/home/ubuntu/code/picongpu/src/cuda_memtest/cuda_memtest.cu:113:413: warning: format ‘%llu’ expects argument of type ‘long long unsigned int’, but argument 6 has type ‘size_t {aka unsigned int}’ [-Wformat=]
     PRINTF("Device name=%s, global memory size=%llu\n", prop->name, prop->totalGlobalMem);
                                                                                                                                                                                                                                                                                                                                                                                                                             ^
/home/ubuntu/code/picongpu/src/cuda_memtest/tests.cu: In function ‘long unsigned int get_random_num_long()’:
/home/ubuntu/code/picongpu/src/cuda_memtest/tests.cu:227:43: warning: left shift count >= width of type [enabled by default]
     unsigned long ret =  ((unsigned long)a) << 32;
                                           ^
Scanning dependencies of target cuda_memtest
[ 83%] Building CXX object build_picongpu/build_cuda_memtest/CMakeFiles/cuda_memtest.dir/misc.cpp.o
Linking CXX executable cuda_memtest
[ 83%] Built target cuda_memtest
/home/ubuntu/code/picongpu/src/libPMacc/include/traits/Limits.tpp(48): error: class "PMacc::traits::limits::Max<size_t>" has already been defined

1 error detected in the compilation of "/tmp/tmpxft_0000028f_00000000-6_main.cpp1.ii".
CMake Error at picongpu_generated_main.cu.o.cmake:264 (message):
  Error generating file
  /home/ubuntu/code/picongpu/build_picongpu/CMakeFiles/picongpu.dir//./picongpu_generated_main.cu.o

make[2]: **\* [build_picongpu/CMakeFiles/picongpu.dir/./picongpu_generated_main.cu.o] Error 1
make[1]: **\* [build_picongpu/CMakeFiles/picongpu.dir/all] Error 2
make: **\* [all] Error 2

--- a/src/libPMacc/include/traits/Limits.tpp
+++ b/src/libPMacc/include/traits/Limits.tpp
@@ -34,12 +34,6 @@ namespace limits
 {
 
 template<>
-struct Max<size_t>
-{
-    static const size_t value=static_cast<size_t>(-1);
-};
-
-template<>
 struct Max<int>
 {
     static const int value=INT_MAX;
@@ -49,6 +43,12 @@ template<>
 struct Max<uint32_t>
 {
     static const uint32_t value=static_cast<uint32_t>(-1);
+};
+
+template<>
+struct Max<uint64_t>
+{
+    static const uint64_t value=static_cast<uint64_t>(-1);
 };
 
 } //namespace limits

